cli/cliutil/awscheck.go

```diff
@@ -0,0 +1,88 @@
+package cliutil
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/netip"
+	"time"
+
+	"golang.org/x/xerrors"
+)
+
+const awsIPRangesURL = "https://ip-ranges.amazonaws.com/ip-ranges.json"
+
+type AWSIPv4Prefix struct {
+	Prefix             string `json:"ip_prefix"`
+	Region             string `json:"region"`
+	Service            string `json:"service"`
+	NetworkBorderGroup string `json:"network_border_group"`
+}
+
+type AWSIPv6Prefix struct {
+	Prefix  string `json:"ipv6_prefix"`
+	Region  string `json:"region"`
+	Service string `json:"service"`
+}
+
+type AWSIPRanges struct {
+	SyncToken    string          `json:"syncToken"`
+	CreateDate   string          `json:"createDate"`
+	IPV4Prefixes []AWSIPv4Prefix `json:"prefixes"`
+	IPV6Prefixes []AWSIPv6Prefix `json:"ipv6_prefixes"`
+}
+
+func NewAWSIPRanges(ctx context.Context) (*AWSIPRanges, error) {
```

Comment:
This needs to be testable with a different URL so we aren't hitting any external servers during testing. You can save an example response from the actual endpoint into a `./testdata/aws-ips-response.json` and load that when making your fake test server